pkg/pipeline/service.go

```diff
@@ -366,3 +336,27 @@ func (s *Service) validatePipeline(cfg Config, id string) error {
 
 	return cerrors.Join(errs...)
 }
+
+// GetInstances returns all pipeline instances.
+func (s *Service) GetInstances() map[string]*Instance {
+	return s.instances
+}
+
+// UpdateStatus updates the status of a pipeline by the ID.
+func (s *Service) UpdateStatus(ctx context.Context, id string, status Status, errMsg string) error {
+	pipeline, err := s.Get(ctx, id)
+	if err != nil {
+		return err
+	}
+	measure.PipelinesGauge.WithValues(strings.ToLower(pipeline.GetStatus().String())).Dec()
```

Comment:
Heads-up - this code changed slightly in the [metrics PR](https://github.com/ConduitIO/conduit/pull/1828), we need to update it. Also, be ready to resolve conflicts from `main` because of this.

Reply:
@lovromazgon thank you for the heads up. I just merged and resolve their conflicts.